### ISSUE TITLE
mpv segmentation fault with certain bitmap subtitles

### ISSUE THREAD

**Issue by gianniqq** [mpv-player/mpv#6136]
### mpv version and platform
using latest release version:
```
Documenti/mpv-build/mpv/build/mpv --version
mpv 0.29.0 Copyright © 2000-2018 mpv/MPlayer/mplayer2 projects
 built on Tue Sep 18 23:06:42 CEST 2018
ffmpeg library versions:
   libavutil       56.19.101
   libavcodec      58.30.100
   libavformat     58.18.101
   libswscale      5.2.100
   libavfilter     7.32.100
   libswresample   3.2.100
ffmpeg version: N-91974-gf4cf6ba
```
platform: KDE Neon 5.13
### Reproduction steps
Reproducing the video with --no-config flag (as well as without it) triggers the segmentation fault:
`mpv --no-config bug_mpv.mkv`

### Expected behavior
mpv is supposed to show subtitles correctly and not crash.

### Actual behavior
Few seconds after the video starts mpv logs some error messages "_[vo/gpu] OSD bitmaps do not fit on a surface with the maximum supported size 2048x2048._": in fact subtitles aren't shown as supposed.
mpv keeps logging those error messages while the subtitles are supposed to be on screen (these are not "simple" subtitles, they have some kind of image background), and when the scene changes and subtitles are supposed to disappear, mpv segfaults. At least, it looks like this is the course of events.
Using --vo=xv doesn't trigger the segmentation fault and correctly displays the subtitles.

### Log file
mpv log file: https://0x0.st/sx0C.txt
mpv backtrace: https://0x0.st/sxGz.txt

### Sample files

sample file: https://0x0.st/sxG-.mkv

**Comment by 2295** [comment]
I am not a professional devepoler so i don't know if this is relevant or obvious, but I noticed that at  `./mpv/video/out/gpu/osd.c:148` there's an if statement that's supposed to check if the texture of the osd bitmap has bounds larger than what is supported by -whatever-, and if that's the case, it displays an error (the "_OSD bitmaps do not fit.._" one seen above) and jumps to the end of the function with a goto:
`./mpv/video/out/gpu/osd.c:152 goto done;`
After that goto there's some code that initializes the texture, and more important, initializes the pointer that seems to cause the seg fault at `./mpv/video/out/gpu/shader_cache.c:316` (at least that's what i interpreted from the backtrace):
`[...]`
**`./mpv/video/out/gpu/osd.c:165 osd->texture = ra_tex_create(ra, &params);`**

I commented the goto and built again mpv, and now **there's no segfault anymore**, but the bitmap doesn't appear (...not that I expected that) and another error appears instead:
```
[vo/gpu] OSD bitmaps do not fit on a surface with the maximum supported size 2048x2048.
[vo/gpu/opengl] after creating texture: OpenGL error INVALID_VALUE.
```

I think I can consider this an improvement, since it doesn't crash anymore and just fails to display the bitmap, but I don't know if I was really supposed to remove that goto and let that few lines of code run.
I hope I helped and look after someone that actually knows what's up.

**Comment by gianniqq** [comment]
I just checked and the segmentation fault does still happen using latest release version:
```
$ ./mpv/build/mpv --version
mpv 0.29.1 Copyright © 2000-2018 mpv/MPlayer/mplayer2 projects
 built on Wed Feb  6 14:24:55 CET 2019
ffmpeg library versions:
   libavutil       56.26.100
   libavcodec      58.46.100
   libavformat     58.26.100
   libswscale      5.4.100
   libavfilter     7.48.100
   libswresample   3.4.100
ffmpeg version: N-93084-g835ab35ef0

```
platform: Manjaro Linux

log file of "`mpv --no-config --msg-level=all=debug --log-file=~/log_mpv.txt https://0x0.st/sxG-.mkv`": https://0x0.st/zsxe.txt

**Comment by haasn** [comment]
A fix for the segfault is already available as 13759d (part of my [libplacebo PR](https://github.com/mpv-player/mpv/pull/6476)).

It won't fix the fact that the subtitles are not rendered, but it will at least prevent mpv from crashing.

**Comment by gianniqq** [comment]
> A fix for the segfault is already available as [13759d5](https://github.com/mpv-player/mpv/commit/13759d56a9f5279e6cc574374ad2235e14c39616) (part of my [libplacebo PR](https://github.com/mpv-player/mpv/pull/6476)).
> 
> It won't fix the fact that the subtitles are not rendered, but it will at least prevent mpv from crashing.

That's great! I hope it will get merged soon

**Comment by Akemi** [comment]
if this issue still persists, please open a new issue.
